perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -61,53 +170,47 @@ def GetInfo():
 def Prepare(benchmark_spec):
   """Prepare the virtual machine to run FIO.
 
-     This includes installing fio, bc. and libaio1 and insuring that the
-     attached disk is large enough to support the fio benchmark.
+     This includes installing fio, bc, and libaio1 and insuring that
+     the attached disk is large enough to support the fio
+     benchmark. We also make sure the job file is always located at
+     the same path on the local machine.
 
   Args:
     benchmark_spec: The benchmark specification. Contains all data that is
         required to run the benchmark.
+
   """
-  vms = benchmark_spec.vms
-  vm = vms[0]
+
+  if FLAGS.fio_jobfile:
+    ignored_flags = []
+    if FLAGS.against_device:
+      ignored_flags.append('--against_device')
+    if FLAGS.device_fill_size != '100%':
+      ignored_flags.append('--device_fill_size')
+    if FLAGS.io_depths != '1':
+      ignored_flags.append('--io_depths')
+
+    if ignored_flags:
+      logging.warning('Fio job file specified. Ignoring options "%s"',
+                      ', '.join(ignored_flags))
+  if FLAGS.device_fill_size != '100%' and not FLAGS.against_device:
```

Comment:
I changed it this morning so we won't prewarm if the user supplies a job file. I think that makes sense to give the user the most flexibility. They can always put a prewarm job in their job file if they want to.